### PR TITLE
core/client/example: auto-request funds if needed

### PR DESCRIPTION
**Pull request by jchappelow**
This updates the `core/client` example with several changes:
- tidy go.mod for current deps
- auto-generate a private key the one in the code is blank, making the example more easily runnable
- if balance is empty or dust, try to request funds from the faucet.  This part works, but is not more generally applicable if say they change the chain.  I might put this handling behind a boolean set in the `var` block next to the other settings. Note that https://github.com/kwilteam/kwil-db/pull/991 will add to the `chain_info` response a flag indicating if the network has gas enabled.  This can be helpful to direct logic depending on whether the network requires gas for transaction.
- fix the incorrect `records.ExportString` method call.  The `Call` method now returns a results struct containing records rather than the records themselves, so we should do `result.Records.ExportString`

### CODE REVIEW

**Comment by Yaiba** [comment]
I guess `core/gatewayclient` should also update the go.mod?

**Comment by jchappelow** [comment]
> I guess `core/gatewayclient` should also update the go.mod?

Ah! I forgot about that example.

**Review by brennanjl — DISMISSED** [review]
Aside from @Yaiba's feedback, lgtm

File: core/gatewayclient/example/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+mykgwapp
```
[the remark this answers is not in the document]
Comment (jchappelow) [reply core/gatewayclient/example/.gitignore:1]: The go.mod sets this name, so the binary gets it too

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -145,6 +145,10 @@ jobs:
           task build:cli
           task build:admin
 
+      - name: compile the core/client/example app
+        run: GOWORK=off go build -o /dev/null
+        working-directory: core/client/example
+
```
[the remark this answers is not in the document]
Comment (jchappelow) [reply .github/workflows/ci.yaml:151]: Yup will do that too

File: core/gatewayclient/example/go.mod [hunk 3]
```diff
@@ -1,21 +1,22 @@
 module mykgwapp
 
-go 1.21.0
+go 1.22
 
-require github.com/kwilteam/kwil-db/core v0.1.1-0.20240514193332-c61ba10f8a1c
+require github.com/kwilteam/kwil-db/core v0.3.0-beta.0.20240917145512-e6922733fc1a
```
Comment (Yaiba) [inline core/gatewayclient/example/go.mod:5]: I see core/client uses a different version, should we keep them same?

**Review by Yaiba — APPROVED** [review]
LGTM, just have a question about the go.mod version

File: core/gatewayclient/example/go.mod (hunk 3 above)
Comment (jchappelow) [reply core/gatewayclient/example/go.mod:5]: Yup, totally.  updated.

**Pull request merged**